consensus/polybft/polybft.go

```diff
@@ -360,6 +360,35 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 				contracts.NativeERC20TokenContract, input, "NativeERC20", transition); err != nil {
 				return err
 			}
+
+			// initialize EIP1559Burn SC
+			if config.Params.BurnContract != nil &&
+				len(config.Params.BurnContract) == 1 &&
+				!polyBFTConfig.NativeTokenConfig.IsMintable {
+				var contractAddress types.Address
+				for _, address := range config.Params.BurnContract {
```

Comment:
Why do we need to maintain this as a map if there's only a single Burn contract?

Reply:
In absence of on chain governance, this is the way how one could configure different burn contracts in the runtime: by manually editing `genesis.json` and restarting the node(s).